snapd-glib/requests/snapd-json.c

```diff
@@ -791,6 +799,26 @@ _snapd_json_parse_change (JsonNode *node, GError **error)
         JsonObject *progress = _snapd_json_get_object (object, "progress");
         g_autoptr(GDateTime) spawn_time = _snapd_json_get_date_time (object, "spawn-time", NULL);
         g_autoptr(GDateTime) ready_time = _snapd_json_get_date_time (object, "ready-time", NULL);
+        JsonObject *data = _snapd_json_get_object (object, "data");
+        g_autoptr(SnapdTaskData) task_data = NULL;
+        if (data != NULL) {
+            JsonNode *affected_snaps_node = json_object_get_member (data, "affected-snaps");
+            if (affected_snaps_node != NULL) {
+                if (json_node_get_value_type (affected_snaps_node) != JSON_TYPE_ARRAY) {
+                    g_set_error (error,
+                                SNAPD_ERROR,
+                                SNAPD_ERROR_READ_FAILED,
+                                "Unexpected change type");
```

Comment:
Copy-paste error - the text doesn't match the error.

Reply:
fixed.